nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/StandardNiFiServiceFacade.java

```diff
@@ -3332,6 +3336,38 @@ private ProcessorEntity createProcessorEntity(final ProcessorNode processor, fin
             .collect(Collectors.toSet());
     }
 
+    @Override
+    public ProcessorsRunStatusDetailsEntity getProcessorsRunStatusDetails(final Set<String> processorIds, final NiFiUser user) {
+        final List<ProcessorRunStatusDetailsEntity> summaryEntities = processorIds.stream()
+            .map(processorDAO::getProcessor)
+            .map(processor -> createRunStatusDetailsEntity(processor, user))
+            .collect(Collectors.toList());
+
+        final ProcessorsRunStatusDetailsEntity summariesEntity = new ProcessorsRunStatusDetailsEntity();
+        summariesEntity.setRunStatusDetails(summaryEntities);
+        return summariesEntity;
+    }
+
+    private ProcessorRunStatusDetailsEntity createRunStatusDetailsEntity(final ProcessorNode processor, final NiFiUser user) {
+        final RevisionDTO revision = dtoFactory.createRevisionDTO(revisionManager.getRevision(processor.getIdentifier()));
+        final PermissionsDTO permissions = dtoFactory.createPermissionsDto(processor, user);
+        final ProcessorStatus processorStatus = controllerFacade.getProcessorStatus(processor.getIdentifier());
+        final ProcessorRunStatusDetailsDTO runStatusDetailsDto = dtoFactory.createProcessorRunStatusDetailsDto(processor, processorStatus);
+
+        if (!Boolean.TRUE.equals(permissions.getCanRead())) {
```

Comment:
Minor: Not sure why `getCanRead()` has the return type `Boolean` when it's actually a getter for a `boolean` property.

Simply doing
```suggestion
        if (permissions.getCanRead()) {
```
should be safe.

Reply:
@tpalfy DTO objects in nifi always use Objects rather than primitives because a primitive would default to `false` and we don't want that - we want the exact value that was set, including `null` potentially. Because it can be `null`, we cannot use auto-boxing, as it could throw NPE

Reply:
Actually, in this case I guess it doesn't matter - we know that the permissions always will be populated. So even though it's a Boolean, it's safe to assume that the value will be set.